src/internal.rs

```diff
@@ -408,7 +401,7 @@ pub struct Rav1dFrameContext_frame_thread {
     /// Indexed using `t.by * f.b4_stride + t.bx`.
     pub b: Vec<Av1Block>,
 
-    pub cbi: Vec<CodedBlockInfo>,
+    pub cbi: Vec<[i16; 3]>, /* bits 0-4: txtp, bits 5-15: eob */
```

Comment:
We should probably newtype `i16` as `CodedBlockInfo`.  The bit-twiddling everywhere is pretty error-prone and brittle.  I can push a commit to do that if that's okay; I tested it out.

Reply:
How does 7375ba3 look?

Reply:
> How does [7375ba3](https://github.com/memorysafety/rav1d/commit/7375ba3f3b4dc5160996606f7c3ba07ba113b0fa) look?

LGTM. I guess that, unlike in C, `<<` is not UB for signed integers.